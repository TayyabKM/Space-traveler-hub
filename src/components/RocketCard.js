import React from 'react';
import {
  Badge, Button, Image, Stack,
} from 'react-bootstrap';
import { useDispatch } from 'react-redux';
import { toggleRocketBooking } from '../redux/rockets/rockets';
// eslint-disable-next-line
const RocketCard = ({ id, name, description, imgUrl, reserved }) => {
  const dispatch = useDispatch();
  const handleClick = () => {
    dispatch(toggleRocketBooking(id));
  };
  return (
    <Stack direction="horizontal" gap={2} className="p-2">
      <div>
        <Image src={imgUrl} alt="rocket" width={250} />
      </div>
      <Stack>
        <h2>{name}</h2>
        {reserved ? (
          <div>
            <Badge bg="success" pill>
              Reserved
            </Badge>
          </div>
        ) : null}
        <p>{description}</p>
        <div>
          {reserved ? (
            <Button variant="outline-danger" onClick={handleClick}>
              Cancel Reserve
            </Button>
          ) : (
            <Button variant="primary" onClick={handleClick}>
              Reserve Rocket
            </Button>
          )}
        </div>
      </Stack>
    </Stack>
  );
};
export default RocketCard;
