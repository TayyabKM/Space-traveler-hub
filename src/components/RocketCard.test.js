import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import RocketCard from './RocketCard';

describe('Component for Individual Rocket', () => {
  it('should render correctly', () => {
    const component = TestRenderer.create(
      <Provider store={store}>
        <RocketCard />
      </Provider>,
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});
