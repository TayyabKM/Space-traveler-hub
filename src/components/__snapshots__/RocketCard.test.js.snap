// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component for Individual Rocket should render correctly 1`] = `
<div
  className="p-2 hstack gap-2"
>
  <div>
    <img
      alt="rocket"
      className=""
      width={250}
    />
  </div>
  <div
    className="vstack"
  >
    <h2 />
    <p />
    <div>
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Reserve Rocket
      </button>
    </div>
  </div>
</div>
`;
